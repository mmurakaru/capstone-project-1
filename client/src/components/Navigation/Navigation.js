import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Navigation.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.string.isRequired,
}

export default function Navigation({ onClick, children }) {
  return <NavButton onClick={onClick}>{children}</NavButton>
}

const NavButton = styled.button`
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  color: var(--primary-dark);
  font-size: 0.8em;
  padding: 6px;
`
