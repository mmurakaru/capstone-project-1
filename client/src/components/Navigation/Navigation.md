Navigation

```js
<Navigation onClick={() => {}}>all songs</Navigation>
```
