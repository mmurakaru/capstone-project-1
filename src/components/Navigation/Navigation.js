import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Navigation.propTypes = {
  handleClick: PropTypes.func,
  children: PropTypes.string.isRequired,
}

export default function Navigation({ handleClick, children }) {
  return <NavButton onClick={handleClick}>{children}</NavButton>
}

const NavButton = styled.button`
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  color: var(--primary-dark);
  padding: 6px;
  font-size: 0.8em;
`
