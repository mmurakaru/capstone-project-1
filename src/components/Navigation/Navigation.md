Navigation Button

```js
<Navigation onClick={() => {}}>all songs</Navigation>
```
