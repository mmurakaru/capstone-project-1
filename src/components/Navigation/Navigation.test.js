import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import Navigation from './Navigation'

describe('Navigation', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Navigation onClick={() => {}}>all songs</Navigation>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('calls onClick correctly', () => {
    const handleClickMock = jest.fn()
    const { getByRole } = render(
      <Navigation handleClick={handleClickMock} isSongPlaying={false}>
        all songs
      </Navigation>
    )
    const button = getByRole('button')
    user.click(button)

    expect(handleClickMock).toHaveBeenCalled()
  })
})
