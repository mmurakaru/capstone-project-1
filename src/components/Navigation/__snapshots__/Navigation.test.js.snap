// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
.c0 {
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  color: var(--primary-dark);
  font-size: 0.8em;
  padding: 6px;
}

<button
  class="c0"
>
  all songs
</button>
`;
